element-connector/src/main/java/org/eclipse/californium/elements/tcp/TcpServerConnector.java

```diff
@@ -82,17 +97,33 @@ public TcpServerConnector(InetSocketAddress localAddress, int numberOfThreads, i
 				.option(ChannelOption.AUTO_READ, true).childHandler(new ChannelRegistry());
 
 		// Start the server.
-		bootstrap.bind(localAddress).syncUninterruptibly().addListener(new ChannelFutureListener() {
-
-			@Override
-			public void operationComplete(ChannelFuture future) throws Exception {
-				InetSocketAddress listenAddress = (InetSocketAddress) future.channel().localAddress();
-				listenUri = getListenUri(listenAddress);
+		ChannelFuture channelFuture = bootstrap.bind(localAddress).syncUninterruptibly();
+
+		if (0 == localAddress.getPort()) {
+			// get assigned port
+			channelFuture.addListener(new ChannelFutureListener() {
```

Comment:
Any value in listening asynchronously, since you already call syncUninterruptibly above?